src/main/java/io/goobox/sync/storj/App.java

```diff
@@ -47,6 +50,8 @@
 
     private static App instance;
 
+    private static int NUM_THREADS = Optional.of(Runtime.getRuntime().availableProcessors()).or(1);
```

Comment:
The JavaDoc of `Runtime.availableProcessors()` reads:

> the maximum number of processors available to the virtual machine; never smaller than one

Do we need the `Optional` helper in this case?

Reply:
for slow internet connection with 8 cpus, and for a few big size files upload, this may create a network latency situation. nevertheless, this is future problem. for now, i just remove this optional class